sbpy/bib/core.py

```diff
@@ -29,8 +29,26 @@ def register(task, citations):
     """
     global _bibliography, _track
     if _track:
-        _bibliography[task] = _bibliography.get(task, {})
-        _bibliography[task].update(citations)
+        if task in _bibliography:
+            key = []
+            value = set()
+            key.append(list(citations.keys()))
+            for i in range(len(key)):
+                if list(citations.keys())[i] in _bibliography[task]:
```

Comment:
Instead of working with lists that are then converted to sets, you can take advantage of sets functionality immediately (e.g., adding the same element to the list will not change it). You just have to distinguish between the case that the subtask already exists or not. Can you fill in the `...` parts? Only a single line is required in each case:

> if newsubtask in _bibliography[task].keys():
> ...
> else:
> ....